jax/_src/interpreters/partial_eval.py

```diff
@@ -1857,6 +1857,10 @@ def process_primitive(self, primitive, tracers, params):
   def default_process_primitive(self, primitive, tracers, params):
     avals = [t.aval for t in tracers]
     out_avals, effects = primitive.abstract_eval(*avals, **params)
+    if not (isinstance(out_avals, (tuple,list)) == primitive.multiple_results):
```

Comment:
can you split this apart into:

`if not isinstance(out_avals, (tuple, list)) and primitive.multiple_results:

Reply:
Can't you just replace `==` with `and`?

Reply:
No. Here is the table of those two operation. When both parameter are false, the results isn't the one we need.
```
              , ==, and
true, true,  true, true
true, false, false, false
false, true, false, false 
false, false, true, false
```

Reply:
I would leave a comment here saying you're effectively using `==` as an XOR.

Reply:
Good idea. I said it is a not xor.